### ISSUE TITLE
[omnisci] Various problems in datetime processing

### ISSUE THREAD

**Issue by gshimansky** [modin-project/modin#3480]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Ubuntu 20.04.1 LTS

- **Modin version** (`modin.__version__`):

0.10.2+39.g15f168ca

- **Python version**:

Python 3.9.7

- **Code we can use to reproduce**:

```python
import modin.pandas as pd

df = pd.read_csv("datetime.csv", parse_dates=["pickup_datetime", "dropoff_datetime"])

print(df.dtypes)
print(df["pickup_datetime"].astype("int64"))

df["dropoff_datetime"][0] = pd.to_datetime("2014-01-09 20:52:31")
df["dropoff_datetime"][1] = pd.to_datetime("2014-01-09 20:55:12")
# If you uncomment this line you get a different exception
# df["dropoff_datetime"][3] = pd.to_datetime("2014-01-09 20:59:46")

print(df["dropoff_datetime"].astype("int64"))
```

### Describe the problem

Code above finds three different problems:
1. When datetime is read from CSV file, it is kept in seconds, not in nanoseconds as type `datetime64[ns]` implies. When this code runs on Pandas it shows numbers like `1389300325000000000` while the same code on Modin on omnisci produces numbers like `1389300325`.
2. When datetime is assigned to column from `pd.to_datetime` it produces an `AssertionError: unaligned dtypes (dropoff_datetime    int64`.
3. If you uncomment a 3rd assignment line this code produces another exception `ValueError: cannot reindex from a duplicate axis`.

Attached is the data file. [datetime.csv](https://github.com/modin-project/modin/files/7227994/datetime.csv)

**Comment by Garra1980** [comment]
What else do we need here?

**Comment by Garra1980** [comment]
Let’s close this then

**Comment by YarShev** [comment]
@ienkovich , did #3483 fully covered this issue?

**Comment by ienkovich** [comment]
I worked on the second problem only.

**Comment by YarShev** [comment]
Let's put the rest part to be resolved in 0.13 then.

**Comment by YarShev** [comment]
HDK engine is deprecated and will be removed in a future version.
